scripts/test/Muon/elemental_analysis_2/elemental_analysis_loadrun_model_test.py

```diff
@@ -0,0 +1,75 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import unittest
+from unittest import mock
+
+from Muon.GUI.ElementalAnalysis2.context.context import RunObject
+from mantidqt.utils.qt.testing import start_qapplication
+
+from Muon.GUI.ElementalAnalysis2.load_widget.load_models import LoadRunWidgetModel
+from Muon.GUI.Common.test_helpers.context_setup import setup_context_for_ea_tests
+
+
+@start_qapplication
+class LoadRunWidgetModelEATest(unittest.TestCase):
+    def setUp(self):
+        setup_context_for_ea_tests(self)
+        self.model = LoadRunWidgetModel(self.loaded_data, self.context)
+
+    def test_model_initialized_with_empty_lists_of_loaded_data(self):
+        self.assertEqual(self.model._current_run, None)
+        self.assertEqual(self.model._directory, "")
+        self.assertEqual(self.model._runs, [])
+        self.assertEqual(self.model._loaded_data_store.get_data(), None)
+        self.assertEqual(self.model._data_context.run_info, [])
+
+    def test_executing_load_without_runs_does_nothing(self):
+        self.model.execute()
+        self.assertEqual(self.model._current_run, None)
+        self.assertEqual(self.model._directory, "")
+        self.assertEqual(self.model._runs, [])
+        self.assertEqual(self.model._loaded_data_store.get_data(), None)
+        self.assertEqual(self.model._data_context.run_info, [])
+
+    @mock.patch('Muon.GUI.ElementalAnalysis2.load_widget.load_models.LoadElementalAnalysisData')
+    def test_execute_successfully_loads_valid_run_single(self, load_ea_data_mock):
+        # Mock the load algorithm
+        grpws = mock.Mock()
+        grpws.getNames.return_value = ['1234; Detector 1', '1234; Detector 3']
+        mocked_path = mock.NonCallableMock()
+        load_return_vals = grpws, mocked_path
+
+        load_ea_data_mock.return_value = load_return_vals
+
+        self.model._runs = [1234]
+        self.model.execute()
+        self.assertEqual(self.model._directory, mocked_path)
+        for run_info in self.model._data_context.run_info:
```

Comment:
how many are we expecting in this loop?

Reply:
In this instance just the 1. self.model._data_context.run_info is a list so presumably even if there is only one value in it we still need to use a loop?

Reply:
if we know its one item we could use `run _info = self.model._data_context.run_info[0]`